rust/arrow/src/ipc/writer.rs

```diff
@@ -349,8 +350,9 @@ fn write_array_data(
     let null_buffer = match array_data.null_buffer() {
         None => {
             // create a buffer and fill it with valid bits
-            let buffer = MutableBuffer::new(num_rows);
-            let buffer = buffer.with_bitset(num_rows, true);
+            let num_bytes = bit_util::ceil(num_rows, 8);
```

Comment:
@andygrove @pauldix @maxburke this might be of interest for interop/integration with other implementations. I was overstating the length of the null buffer by 8. I suspect that with a sufficiently large batch size (like the 1 million that I tried with), it could cause issues with reading rust-generated IPC files.